ts/Series/GeoHeatmap/GeoHeatmapSeries.ts

```diff
@@ -175,11 +281,418 @@ class GeoHeatmapSeries extends MapSeries {
+                    projectedPixelData[i + 1] =
+                        cartesianImageData.data[redPos + 1];
+                    projectedPixelData[i + 2] =
+                        cartesianImageData.data[redPos + 2];
+                    projectedPixelData[i + 3] =
+                        cartesianImageData.data[redPos + 3];
+                }
+            }
+        }
+
+        return projectedPixelData;
+    }
+
+    public searchPoint(
+        e: PointerEvent,
+        compareX?: boolean | undefined
+    ): Point | undefined {
+        const series = this,
+            chart = this.chart,
+            mapView = chart.mapView;
+
+        if (
+            mapView &&
+            series.bounds &&
+            series.image &&
+            chart.tooltip &&
+            chart.tooltip.options.enabled
+        ) {
+            if (
+                // If user drags map do not build k-d-tree
+                chart.pointer.hasDragged === false &&
+                // If user zooms in/out map do not build k-d-tree
+                (
+                    +series.image.attr('animator') <= 0.01 ||
+                    +series.image.attr('animator') >= 0.99
+                )
+            ) {
+                const topLeft = mapView.projectedUnitsToPixels({
+                        x: series.bounds.x1,
+                        y: series.bounds.y2
+                    }),
+                    bottomRight = mapView.projectedUnitsToPixels({
+                        x: series.bounds.x2,
+                        y: series.bounds.y1
+                    });
+                chart.pointer.normalize(e);
+
+                if (
+                    e.lon && e.lat &&
+                    topLeft && bottomRight &&
+                    e.chartX - chart.plotLeft > topLeft.x &&
+                    e.chartX - chart.plotLeft < bottomRight.x &&
+                    e.chartY - chart.plotTop > topLeft.y &&
+                    e.chartY - chart.plotTop < bottomRight.y
+                ) {
+                    return this.searchKDTree({
+                        clientX: e.chartX,
+                        lon: normalizeLonValue(e.lon),
+                        lat: e.lat
+                    }, compareX, e);
+                }
+            } else {
+                chart.tooltip.destroy();
+            }
+        }
+    }
 }
 
+addEvent(GeoHeatmapSeries, 'afterDataClassLegendClick', function (): void {
+    this.isDirtyCanvas = true;
+    this.drawPoints();
+});
+
+// Modify final series options.
+addEvent(GeoHeatmapSeries, 'afterSetOptions', function (
+    e: { options: GeoHeatmapSeriesOptions }
+): void {
+    const options = e.options;
+
+    if (!isObject(options.interpolation)) {
+        // Create the extended object out of the boolean
+        options.interpolation = merge(
+            true,
+            GeoHeatmapSeries.defaultOptions.interpolation,
+            {
+                enabled: options.interpolation
+            }
+        );
+    }
+});
```

Comment:
I have created the extended object out of the boolean by using this event. I am not sure if that's an acceptable solution @TorsteinHonsi, but works pretty well.

Reply:
This doesn't seem to work well with `series.update`. In the earth statistics demo, try adding this after chart creation:

```js
    setTimeout(function () {
        chart.series[0].update({
            interpolation: true
        });
    }, 1234);
```

I see that the `update` function merges the new, raw options into the existing options, so there you will get the boolean for of interpolation. Maybe you should just have a `getInterpolation` method that handles this, so you can do `this.getInterpolation().enabled` instead of `this.options.interpolation.enabled`. Performance wise it wouldn't matter since it's only called 2 or 3 times per render.

Another thing, when doing `merge` with the first argument set to true, we actually modify the default options, which is not what we want. Once we remove the `true` argument, I think this merge is okay.

Reply:
Added `getInterpolation` method - looks solid for me now.